File: engine/model/src/main/scala/net/server/SkillMacro.scala
package net.server

class SkillMacro(private var _name: String, private var _shout: Int, private var _skill1: Int, private var _skill2: Int, private var _skill3: Int, private var _position: Int) {
  def name: String = _name

  def shout: Int = _shout

  def skill1: Int = _skill1

  def skill2: Int = _skill2

  def skill3: Int = _skill3

  def position: Int = _position
}

File: engine/engine-model/src/main/java/net/server/SkillMacro.java
package net.server;

public record SkillMacro(String name, Integer shout, Integer skill1, Integer skill2, Integer skill3, Integer position) {
   public SkillMacro setSkill1(Integer skill1) {
      return new SkillMacro(name, shout, skill1, skill2, skill3, position);
   }

   public SkillMacro setSkill2(Integer skill2) {
      return new SkillMacro(name, shout, skill1, skill2, skill3, position);
   }

   public SkillMacro setSkill3(Integer skill2) {
      return new SkillMacro(name, shout, skill1, skill2, skill3, position);
   }
}
